pkg/pillar/cmd/volumemgr/handlers.go

```diff
@@ -0,0 +1,640 @@
+// Copyright (c) 2022 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package volumemgr
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/lf-edge/edge-containers/pkg/registry"
+	zconfig "github.com/lf-edge/eve/api/go/config"
+	"github.com/lf-edge/eve/pkg/pillar/cas"
+	"github.com/lf-edge/eve/pkg/pillar/containerd"
+	"github.com/lf-edge/eve/pkg/pillar/diskmetrics"
+	"github.com/lf-edge/eve/pkg/pillar/tgt"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	utils "github.com/lf-edge/eve/pkg/pillar/utils/file"
+	"github.com/lf-edge/eve/pkg/pillar/zfs"
+)
+
+type volumeHandler interface {
+	// prepareVolume to be created
+	prepareVolume() error
+	// checkAndHandleCreated handles preparation done
+	checkAndHandlePrepared() (bool, error)
```

Comment:
What is the purpose of separation of:
 * prepare into `prepareVolume` and `checkAndHandlePrepared`
 * create into `createVolume` and `checkAndHandleCreated`
?

Reply:
We run such function from different place, prepareVolume from async worker, checkAndHandlePrepared from the main routine. The first one is about actual creating the second one is about updating fields after creation is done. Both of them depends on volume type so I decided to move them into interface,